### ISSUE TITLE
Support for Moving Geometry in Unreal Plug-in

### ISSUE THREAD

**Issue by DavidWLejeune** [ValveSoftware/steam-audio#108]
While Moving Geometry has been added to the Unity plug-in, it has still not been added to the Unreal plug-in. The Unity and Unreal plug-ins should have feature parity.

**Comment by 1runeberg** [comment]
Hi @DavidWLejeune,  

Thanks for your feedback. We are working on this currently. I'll post an update here once we have some code we can share publicly.

Cheers,
Rune

**Comment by 1runeberg** [comment]
Hi @DavidWLejeune -  we've published a PR dynamic geometry here here:
https://github.com/EpicGames/UnrealEngine/pull/6538

This brings the UE plugin closer to Unity plugin parity, but added a little extra for UE4 devs - you can independently manipulate transforms of SMCs in a single Actor as shown in this demo vid using Paragon assets:

https://www.youtube.com/watch?v=VtJ3_jcBpcs&feature=youtu.be

Enjoy, feedback welcome :)

**Comment by DavidWLejeune** [comment]
Awesome! We'll try integrating into our in-house branch asap to give it a test. We're on a 4.23 base, so fingers crossed it won't give us too much trouble.

**Comment by 1runeberg** [comment]
Cool. let me know. This is on top of 4.24 changes including the update SteamAudio API, so you may need to pull those in as well.

**Comment by DavidWLejeune** [comment]
So I'm running into an issue with geo that is in a sub-level: If I move the geo then hide the sub-level, the acoustics of the scene update correctly. But if I unhide the sub-level and move the geo back to where it was, nothing changes. (Also if I don't move geo at all. Just hiding and unhiding sublevels, the dynamic geo doesn't do anything, but I do see the "Dynamic Geometry Added" logging).

Can it handle having the same mesh instance creation logic that's in InitializeEnvironment() in the AddDynamicGeometry() function in PhononPluginManager? Or would I be barking up the wrong tree with that?

**Comment by DavidWLejeune** [comment]
Update on this: I was able to handle level visibility changes by moving all the DynamicScene stuff in InitializeEnvironment into its own function (which I named UpdateDynamicScene). I then call that function both from InitializeEnvironment and AddDynamicGeometry (which you were already calling from BeginPlay on the PhononGeometryComponent).

Destroy I wasn't able to properly solve, so I did a hack where I just send a transform with a scale of 0 to iplUpdateInstancedMeshTransform in RemoveDynamicGeometry.
